### PR TITLE
Updating destination waypoint when route changes

### LINKED ISSUE #1326: Destination annotation fails to update when route changes
**Mapbox Navigation SDK version: 0.16.2**

### Steps to trigger behavior

 1. Create a new view controller that displays a navigation route between a user's location and a fixed destination.
 2. Add a `UIButton` that triggers an `IBAction` that calculates a new route to a new destination.
 [_Full minimal test case **here.**_](https://github.com/mapbox/navigation-ios-examples/blob/nadia-156562/Navigation-Examples/Examples/Basic.swift#L67-L69)
 3. Observe issue: the destination annotation fails to move to the new destination, even after the route line has updated.

### Expected behavior
- The destination annotation should update to the new destination if the route is changed.
### Actual behavior
- The destination annotation does not update to the new destination if the route is changed, as shown below:

navbug

### PR DESCRIPTION
**Pull request by JThramer**
Fixes #1326. Issue was caused by a missing `showWaypoints(route:)` call.

It's worth calling out that the order-of-operations for the route that the NVC gets initialized with is _really_ obscure and brittle. For instance, unless I manually do `navigationViewController.route = initialRoute` _after_ I wait for the styles to load, I cannot find any annotations on the map, which doesn't seem like reasonable behavior. I thought it out of scope for this issue, but I'll cut a ticket for this as I close this PR out.

/cc @mapbox/navigation-ios

### CODE REVIEW

**Comment by bsudekum** [comment]
It took me a second to understand the link between notifyDidReroute and why this would fix #1326, but then I realized when route is set:

https://github.com/mapbox/mapbox-navigation-ios/blob/026fddd1bc07080f33d0cf3f5e7df3b41a04781a/MapboxNavigation/NavigationViewController.swift#L216-L227

**Comment by JThramer** [comment]
There was an attempt.

Posterity: Apparently my green lights weren't green enough. Holding off on merging this until we can figure out why my tests were passing locally and yet not on CI.

It's also a bit disturbing that results from local testing and CI are not consistent. We should cut a ticket to investigate this and hopefully eliminate any inconsistancies.

**Comment by akitchen** [comment]
> It's also a bit disturbing that results from local testing and CI are not consistent. We should cut a ticket to investigate this and hopefully eliminate any inconsistancies.

These are new tests I believe, so let's get them working properly before merging this PR.

If I'm not mistaken, this new test is loading a style over the network. That's something we want to avoid -- please load one locally from disk instead. This test fails for me on a freshly initialized simulator if I'm offline.

File: MapboxNavigationTests/NavigationViewControllerTests.swift [hunk 1]
```diff
@@ -187,3 +233,35 @@ extension NavigationViewControllerTests {
...
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("This initalizer is not supported in this testing subclass.")
+    }
+}
+
+class TestableDayStyle: DayStyle {
+    required init() {
+        super.init()
+        mapStyleURL = Fixture.blankStyle
```
Comment (bsudekum) [inline MapboxNavigationTests/NavigationViewControllerTests.swift]: Sort of unfortunate we had to do this deep but, 👍

**Pull request merged**